include/event_dbus_monitor.hpp

```diff
@@ -64,32 +103,129 @@ inline void VMIIPChange(sdbusplus::message::message& msg)
-                    infName;
-                BMCWEB_LOG_DEBUG
-                    << "Pushing the VMI IP change Event with origin : "
-                    << origin;
-                redfish::EventServiceManager::getInstance().sendEvent(
-                    redfish::messages::resourceChanged(), origin,
-                    "EthernetInterface");
-            }
+inline void sendEventOnEthIntf(std::string origin)
+{
+    redfish::EventServiceManager::getInstance().sendEvent(
+        redfish::messages::resourceChanged(), origin, "EthernetInterface");
+}
+
+inline void VMIIPPropertyChange(sdbusplus::message::message& msg)
+{
+    if (msg.is_method_error())
+    {
+        BMCWEB_LOG_ERROR << "BMC Hypervisor IP properties changed Signal error";
+        return;
+    }
+
+    std::string objPath = msg.get_path();
+
+    if ((objPath !=
+         "/xyz/openbmc_project/network/hypervisor/eth0/ipv4/addr0") &&
+        (objPath != "/xyz/openbmc_project/network/hypervisor/eth1/ipv4/addr0"))
+    {
+        return;
+    }
+
+    std::string infName;
+    if (objPath.find("/eth0") != std::string::npos)
+    {
+        infName = "eth0";
+    }
+    else if (objPath.find("/eth1") != std::string::npos)
+    {
+        infName = "eth1";
+    }
+
+    boost::container::flat_map<std::string, std::variant<std::string, uint8_t>>
+        values;
+    std::string objName;
+    msg.read(objName, values);
+
+    bool enabledValue;
+    if (infName == "eth0")
+    {
+        enabledValue = isIPEnabledOnIntfEth0;
+    }
+    else if (infName == "eth1")
+    {
+        enabledValue = isIPEnabledOnIntfEth1;
+    }
+
+    std::string origin =
+        "/redfish/v1/Systems/hypervisor/EthernetInterfaces/" + infName;
+    auto find = values.find("Address");
+    if (find != values.end())
+    {
+        const std::string propValue = std::get<std::string>(find->second);
+        if (enabledValue)
+        {
+            BMCWEB_LOG_DEBUG
+                << "Pushing the VMI IP property change event for Address: "
+                << propValue << " with origin : " << origin;
+            sendEventOnEthIntf(origin);
+        }
+    }
+
+    find = values.find("Gateway");
+    if (find != values.end())
+    {
+        const std::string propValue = std::get<std::string>(find->second);
+        if (enabledValue)
+        {
+            BMCWEB_LOG_DEBUG
+                << "Pushing the VMI IP property change event for Gateway: "
+                << propValue << " with origin : " << origin;
+            sendEventOnEthIntf(origin);
+        }
+    }
+
+    find = values.find("PrefixLength");
+    if (find != values.end())
+    {
+        const int64_t propValue =
+            static_cast<int64_t>(std::get<uint8_t>(find->second));
+        if (enabledValue)
+        {
+            BMCWEB_LOG_DEBUG
+                << "Pushing the VMI IP property change event for PrefixLength: "
+                << propValue << " with origin : " << origin;
+            sendEventOnEthIntf(origin);
```

Comment:
We need not send 3 events for 3 values. Send single event if any of the values are changed.

Reply:
So we listen on the property change signal on the IP interface, on each property change, we hit this method and send an event. This is only in case of dhcp configuration.